nano/node/socket.cpp

```diff
@@ -324,28 +342,43 @@ size_t network_prefix)
 bool nano::server_socket::limit_reached_for_incoming_subnetwork_connections (std::shared_ptr<nano::socket> const & new_connection)
 {
 	debug_assert (strand.running_in_this_thread ());
-	if (node.flags.disable_max_peers_per_subnetwork || nano::transport::is_ipv4_or_v4_mapped_address (new_connection->remote.address ()))
+
+	if (!new_connection->remote_endpoint ().has_value ())
+	{
+		return false;
+	}
+
+	if (node.flags.disable_max_peers_per_subnetwork || nano::transport::is_ipv4_or_v4_mapped_address (new_connection->remote_endpoint ()->address ()))
 	{
 		// If the limit is disabled, then it is unreachable.
 		// If the address is IPv4 we don't check for a network limit, since its address space isn't big as IPv6 /64.
 		return false;
 	}
+
 	auto const counted_connections = socket_functions::count_subnetwork_connections (
 	connections_per_address,
-	new_connection->remote.address ().to_v6 (),
+	new_connection->remote_endpoint ()->address ().to_v6 (),
```

Comment:
I don't think remote_endpoint () is guaranteed to dereference here. While there is a check remote_endpoint ().has_value () above, it's calling remote_endpoint () again here and it may not have a value during this call.